src/CohortManager.js

```diff
@@ -0,0 +1,143 @@
+const Cohort = require('./Cohort.js')
+const Student = require('./Student.js')
+
+class CohortManager {
+  constructor() {
+    this.cohortList = []
+    this.studentId = 1
+  }
+
+  createCohort(name) {
+    this.#checkType(name, 'string', 'must search for a string')
+
+    const { ...newCohort } = new Cohort(name)
+    this.cohortList.push(newCohort)
+    return this.cohortList
+  }
+
+  searchForCohort(searchQuery) {
+    this.#checkType(searchQuery, 'string', 'must search for a string')
+
+    const searchResult = this.#findCohort(searchQuery.toLowerCase())
+
+    if (searchResult) {
+      return searchResult
+    }
+    throw new Error('no match found')
+  }
+
+  removeCohort(cohortName) {
+    this.#checkType(cohortName, 'string', 'must be a string')
+
+    const cohortToRemove = this.#findCohort(cohortName.toLowerCase())
+    this.#throwErrorIfUndefined(cohortToRemove)
+
+    const indexOfCohortToRemove = this.cohortList.indexOf(cohortToRemove)
+    this.cohortList.splice(indexOfCohortToRemove, 1)
+    return 'Removed successfully'
+  }
+
+  createStudent(firstName, lastName, gitHubUsername, email) {
+    const { ...newStudent } = new Student(
+      this.studentId,
+      firstName,
+      lastName,
+      gitHubUsername,
+      email
+    )
+    this.studentId++
+    return newStudent
+  }
+
+  addStudentToCohort(student, cohortName) {
+    this.#checkType(student, 'object', `student must be an object`)
+    this.#checkType(cohortName, 'string', `must be a string`)
+
+    const cohortToAddStudentTo = this.#findCohort(cohortName.toLowerCase())
```

Comment:
Does findCohort require a lower case string to be passed? That's an implementation detail if so, it should be abstracted away by moving the .toLowerCase call to within the findCohort method